.gitlab-ci.yml

```
@@ -126,10 +121,36 @@ bump-version-on-datadog-agent:
     - pushd datadog-agent
     - git checkout -b auto-bump/bump-test-infra-$CI_COMMIT_SHORT_SHA
     - export PREVIOUS_SHA=$(cat .gitlab/common/test_infra_version.yml | grep 'TEST_INFRA_DEFINITIONS_BUILDIMAGES:' | awk -F " " '{print $NF}')
-    - inv -e buildimages.update-test-infra-definitions --commit-sha $CI_COMMIT_SHA
-    - git add test/new-e2e/go.mod test/new-e2e/go.sum .gitlab/common/test_infra_version.yml
+    - inv -e buildimages.update-test-infra-definitions --commit-sha $CI_COMMIT_SHA $EXTRA_UPDATE_ARGS
+    - inv -e tidy
+    - git add -u
     - git commit -m "[test-infra-definitions][automated] Bump test-infra-definitions to $CI_COMMIT_SHORT_SHA"
     - git push -f origin auto-bump/bump-test-infra-$CI_COMMIT_SHORT_SHA
     - popd
     - pip install -r requirements.txt
     - inv ci.create-bump-pr-and-close-stale-ones-on-datadog-agent --branch auto-bump/bump-test-infra-$CI_COMMIT_SHORT_SHA --new-commit-sha $CI_COMMIT_SHA --old-commit-sha $PREVIOUS_SHA
+
+bump-version-on-datadog-agent-dev:
+  extends: .bump-version-on-datadog-agent
+  stage: post-release
+  image: ${CI_REGISTRY_IMAGE_TEST}:${CI_COMMIT_SHORT_SHA}
+  tags: ["arch:amd64"]
+  needs: ["build-runner-image"]
+  rules:
+    - if: $CI_COMMIT_BRANCH == "main"
+      when: never
+    - when: manual
+      allow_failure: true
+  variables:
+    EXTRA_UPDATE_ARGS: "--is-dev-image"
+
+bump-version-on-datadog-agent:
+  extends: .bump-version-on-datadog-agent
+  stage: post-release
+  image: ${CI_REGISTRY_IMAGE_TEST}:${CI_COMMIT_SHORT_SHA}
+  tags: ["arch:amd64"]
+  needs: ["release-runner-image"]
+  rules:
+    - if: $CI_COMMIT_BRANCH == "main"
+      when: on_success
```

What about we keep a single job that is automatic on main and manual on dev branch?
```suggestion
bump-version-on-datadog-agent:
  extends: .bump-version-on-datadog-agent
  stage: post-release
  image: ${CI_REGISTRY_IMAGE_TEST}:${CI_COMMIT_SHORT_SHA}
  tags: ["arch:amd64"]
  needs: ["release-runner-image"]
  rules:
    - if: $CI_COMMIT_BRANCH == "main"
      when: on_success
    - when: manual
       allow_failure: true
       variables:
         EXTRA_UPDATE_ARGS: "--is-dev-image"
```

---

Thought about this, but they would have different `needs`, as the `main` one needs the release image, while the dev one needs the test image. Could not find how to set needs based on rules

---

Updated the suggestion! But that should work as well with needs in rules
Per https://docs.gitlab.com/ee/ci/yaml/#rulesneeds

---

I'm also removing the private job definition, no need for it anymore

---

Done!